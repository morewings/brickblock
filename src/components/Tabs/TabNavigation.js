import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {Nav, NavItem, NavLink} from 'reactstrap';

const TabNavigation = (props) => {
  const createTabClassName = activeTab => classnames({
    active: activeTab === props.activeTab,
  });
  const onClick = tab => () => {
    props.toggleTab(tab);
  };
  return (
    <Nav tabs>
      <NavItem>
        <NavLink
          className={createTabClassName('tab1')}
          onClick={onClick('tab1')}>
          Tab1
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink
          className={createTabClassName('tab2')}
          onClick={onClick('tab2')}>
          Tab 2
        </NavLink>
      </NavItem>
    </Nav>
  );
};

TabNavigation.propTypes = {
  toggleTab: PropTypes.func.isRequired, // eslint-disable-line
  activeTab: PropTypes.string.isRequired,
};

export default TabNavigation;
