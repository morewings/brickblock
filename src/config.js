const config = Object.freeze({
  dataUrl: 'https://blockchain-ico.brickblock.io/inputs',
});

export default config;
