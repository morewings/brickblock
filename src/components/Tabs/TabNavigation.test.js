import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import config from 'config';
import TabNavigation from 'components/Tabs/TabNavigation';
import {NavLink} from 'reactstrap';

const toggleTab = jest.fn();

const defaultProps = {
  toggleTab,
  activeTab: config.defaultTab,
};

describe('components > Tabs > TabNavigation', () => {
  beforeEach(() => {
    toggleTab.mockClear();
  });
  it('renders without crashing', () => {
    const wrapper = shallow(<TabNavigation {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('calls `props.toggleTab` on tab link click with `tabId` as an argument', () => {
    const wrapper = shallow(<TabNavigation {...defaultProps} />);
    const tabIds = [config.defaultTab, 'ico'];
    const nav1 = wrapper.find(NavLink).first();
    const nav2 = wrapper.find(NavLink).last();
    nav1.simulate('click');
    expect(toggleTab).toHaveBeenCalledTimes(1);
    expect(toggleTab).toHaveBeenCalledWith(tabIds[0]);
    nav2.simulate('click');
    expect(toggleTab).toHaveBeenCalledTimes(2);
    expect(toggleTab).toHaveBeenCalledWith(tabIds[1]);
  });
});
