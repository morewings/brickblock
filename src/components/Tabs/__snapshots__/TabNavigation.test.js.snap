// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Tabs > TabNavigation renders without crashing 1`] = `
<Nav
  tabs={true}
  tag="ul"
  vertical={false}
>
  <NavItem
    tag="li"
  >
    <NavLink
      className="active"
      onClick={[Function]}
      tag="a"
    >
      PreICO
    </NavLink>
  </NavItem>
  <NavItem
    tag="li"
  >
    <NavLink
      className=""
      onClick={[Function]}
      tag="a"
    >
      ICO
    </NavLink>
  </NavItem>
</Nav>
`;
